sdk/core/core-rest-pipeline/src/policies/retryPolicy.ts

```diff
@@ -59,10 +59,15 @@ export function retryPolicy(
           logger.info(`Retry ${retryCount}: Received a response from request`, request.requestId);
         } catch (e) {
           logger.error(`Retry ${retryCount}: Received an error from request`, request.requestId);
+
+          // RestErrors are valid targets for the retry strategies.
```

Comment:
I feel that it should also be up to retry strategies to decide whether to retry on non RestErrors.  Or are we sure that those should never be retried?

Reply:
We can extend this in the future, but without a clear scenario I wouldn’t advise preemptively extending it 🤔